### DISCUSSION TITLE
3D model channels

### DISCUSSION THREAD

**Discussion by kristianlindholm** [MIC-DKFZ/nnUNet#574] [Q&A] [answered]
Hi

I've used nnU-Net to train a 3D model and it's working perfect! Thanks for the great work!

Now I'm curious about one thing. As my network was initially not printed as pdf, I got the following output about the structure.

 
2021-03-02 14:31:40.769157: Generic_UNet(
  (conv_blocks_localization): ModuleList(
    (0): Sequential(
      (0): StackedConvLayers(
        (blocks): Sequential(
          (0): ConvDropoutNormNonlin(
            (conv): Conv3d(640, 320, kernel_size=[3, 3, 3], stride=(1, 1, 1), padding=[1, 1, 1])
            (instnorm): InstanceNorm3d(320, eps=1e-05, momentum=0.1, affine=True, track_running_stats=False)
            (lrelu): LeakyReLU(negative_slope=0.01, inplace=True)
          )
....

In the article about nnU-Net it states that the number of feature maps are capped at 320 for 3D networks, so how am I supposed to understand this Conv3D layer which seems to have 640 input channels?

Best
Kristian

**Comment by FabianIsensee** [comment, marked as answer]
Hi,
the 640 denotes the number of input features.
If you concatenate 320 from below and 320 from skip you get 640.
None of the convolutions have more then 320 output features ;-)
Best,
Fabian

**Reply by kristianlindholm** [reply]
Ahhh of course! Completely forgot about the skips when thinking about this!
Thank you very much!

Best,
Kristian
